### DISCUSSION TITLE
Concerning java development, and starting as a newbie.

### DISCUSSION THREAD

**Discussion by danielzdev** [LazyVim/LazyVim#1106] [Q&A]
First off, high everyone! I am relatively new to coding (currently in my data structures class for java) and i have been trying out neovim. I wont lie its really hard (so if anyone wouldn't mind helping i would love you forever), but i was wondering if it would be possible to use lazy-vim for java development, specifically backend work with springboot. I'm sure i butchered about half of those, but i really want to dive in and learn how to use this thing, one part i'm stuck at is getting the lsp for java to work, and also i have no idea how to enable linting. Any help is appreciated, and even if no one answers, just wanted to say that this neovim distro is awesome!

**Comment by briandipalma** [comment]
I do virtually no Java development but I'd start by adding the treesitter config for Java: https://github.com/briandipalma/ansible/blob/main/roles/common/files/nvim/lua/plugins/nvim-treesitter.lua#L23 and installing jdtls with mason: https://github.com/briandipalma/ansible/blob/main/roles/common/files/nvim/lua/plugins/mason.lua#L20 and if you want you can spend time setting up https://github.com/mfussenegger/nvim-jdtls

**Reply by dpetka2001** [reply]
I've been keeping track of some user configs mentioned here in the repo and I found one for Java (though I don't use it myself). You might want to take a look at it to get some ideas how to configure it yourself. [This](https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/extras/lang/java.lua) is it by @appelgriebsch.

**Reply by danielzdev** [reply]
Thanks!

**Comment by josemiguelo** [comment]
Here's another one that might help you in your configuration process https://github.com/imbacraft/dusk.nvim/blob/master/nvim/ftplugin/java.lua

**Reply by danielzdev** [reply]
OMG this looks like what ive been looking for thanks!
